src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/meta/ConstantPool.java

```diff
@@ -165,6 +165,28 @@ interface BootstrapMethodInvocation {
         /**
          * Gets the static arguments with which the bootstrap method will be invoked.
          *
+         * An argument of type {@link PrimitiveConstant} represents a {@code CONSTANT_Dynamic_info}
+         * entry. To resolve this entry, the corresponding bootstrap method has to be called first:
+         *
+         * <pre>
+         * lookupConstant(int index, boolean resolve) {
```

Comment:
This example should focus on resolving the static arguments:
```
List<JavaConstant> args = cp.getStaticArguments();
List<JavaConstant> resolvedArgs = new ArrayList<>(args.size());
for (JavaConstant c : args) {
    JavaConstant r = c;
    if (c instanceof PrimitiveConstant pc) {
        r = ...
    }
    resolvedArgs.append(rc);
}
```
It should also use real Java code to remove all ambiguity.